logback-android/src/main/java/ch/qos/logback/core/android/AndroidContextUtil.java

```diff
@@ -107,7 +107,9 @@ public String getMountedExternalStorageDirectoryPath() {
    * @return the absolute path to the external storage directory
    */
   public String getExternalStorageDirectoryPath() {
-    return Environment.getExternalStorageDirectory().getAbsolutePath();
```

Comment:
Since the deprecation starts at API 29, we should only call that function when `Build.VERSIONS.SDK_INT >= 29` (similar to what's done in `getNoBackupFilesDirectoryPath()`).

Reply:
The thing here is that the alternative is available since API version 8. I'll provide a switch though.